src/doc/tarpl/exotic-sizes.md

```diff
@@ -0,0 +1,118 @@
+% Exotically Sized Types
+
+Most of the time, we think in terms of types with a fixed, positive size. This
+is not always the case, however.
+
+
+
+
+
+# Dynamically Sized Types (DSTs)
+
+Rust also supports types without a statically known size. On the surface, this
+is a bit nonsensical: Rust *must* know the size of something in order to work
+with it! DSTs are generally produced as views, or through type-erasure of types
```

Comment:
This is in particular a reference to slice/str -- technically distinct from type-erasure because of subslicing and Vec (which never had the len in the type). Can clarify.